drivers/mfd/rpi-rp2040-gpio-bridge.c

```diff
@@ -0,0 +1,1155 @@
+		i2c_unlock_bus(client->adapter, I2C_LOCK_ROOT_ADAPTER);
+	return 0;
+}
+
+static int rp2040_gbdg_transfer_bypass(struct rp2040_gbdg *priv_data,
+				       const u8 *data, unsigned int length)
+{
+	int ret;
+
+	if (priv_data->fast_xfer_gpios) {
+		ret = rp2040_gbdg_10byte_cmd(
+			priv_data->client, CMD_DAT_EMIT,
+			priv_data->fast_xfer_recv_gpio_base, length);
+		return ret ? ret :
+			     rp2040_gbdg_fast_xfer(priv_data, data, length);
+	} else {
+		u8 *buf = priv_data->buffer;
+
+		while (length) {
+			unsigned int xfer = min(length, HALF_BUFFER);
+
+			buf[0] = DIRECT_PREFIX;
+			buf[1] = DIRECT_CMD_EMIT;
+			memcpy(&buf[2], data, xfer);
+			ret = rp2040_gbdg_i2c_send(priv_data->client, buf,
+						   xfer + 2);
+			if (ret)
+				return ret;
+			length -= xfer;
+			data += xfer;
+		}
+		return 0;
+	}
+}
+
+static int rp2040_gbdg_transfer_cached(struct rp2040_gbdg *priv_data,
+				       const u8 *data, unsigned int length)
+{
+	int ret;
+
+	// Caching mechanism divides data into '8KiB - 9' (8183 byte)
+	// 'RP2040_GBDG_BLOCK_SIZE' blocks.
+	//
+	// If there's a large amount of data to send, instead, attempt to make use
+	// of a manifest.
+	if (length > (2 * RP2040_GBDG_BLOCK_SIZE)) {
+		if (!rp2040_gbdg_transfer_manifest(priv_data, data, length))
+			return 0;
+	}
+
+	while (length) {
+		unsigned int xfer = min(length, RP2040_GBDG_BLOCK_SIZE);
+
+		ret = rp2040_gbdg_transfer_block(priv_data, data, xfer);
+		if (ret)
+			return ret;
+		length -= xfer;
+		data += xfer;
+	}
+	return 0;
+}
+
+static int rp2040_gbdg_transfer_one(struct spi_controller *ctlr,
+				    struct spi_device *spi,
+				    struct spi_transfer *transfer)
+{
+	/* All transfers are performed in a synchronous manner. As such, return '0'
+	 * on success or -ve on failure. (Returning +ve indicates async xfer)
+	 */
+
+	struct rp2040_gbdg *priv_data = spi_controller_get_devdata(ctlr);
+
+	if (priv_data->bypass_cache) {
+		return rp2040_gbdg_transfer_bypass(priv_data, transfer->tx_buf,
+						   transfer->len);
+	} else {
+		return rp2040_gbdg_transfer_cached(priv_data, transfer->tx_buf,
+						   transfer->len);
+	}
+}
+
+static void rp2040_gbdg_set_cs(struct spi_device *spi, bool enable)
+{
+	// 'enable' is inverted and instead describes the logic level of an
+	// active-low CS.
+	struct rp2040_gbdg *priv_data =
+		spi_controller_get_devdata(spi->controller);
+
+	static const char enable_cs[] = { DIRECT_PREFIX, DIRECT_CMD_CS, 0x10 };
+	static const char disable_cs[] = { DIRECT_PREFIX, DIRECT_CMD_CS, 0x00 };
+
+	rp2040_gbdg_i2c_send(priv_data->client, enable ? disable_cs : enable_cs,
+			     3);
+}
+
+static int rp2040_gbdg_gpio_request(struct gpio_chip *gc, unsigned int offset)
+{
+	struct rp2040_gbdg *priv_data = gpiochip_get_data(gc);
+	u32 pattern;
+	int ret;
+
+	if (offset >= NUM_GPIO)
+		return -EINVAL;
+
+	pattern = (1 << (offset + 8));
+	if (pattern & priv_data->gpio_requested)
+		return -EBUSY;
+
+	// Resume if previously no gpio requested
```

Comment:
Normal `/*   */` instead of `//`
Other places too, but this was the first I noticed.

`checkpatch.pl --strict` isn't a bad policy to adopt.